### ISSUE TITLE
Error when trying to create new project w/ Hapi 6.8.1

### ISSUE THREAD

**Issue by pdehaan** [lefos987/generator-ghurt#16]
### Steps to reproduce
1. `$ yo ghurt:hapi` (and accept all the default settings)
2. `$ npm start`
### Actual results

``` sh
$ npm start

> foo@0.0.1 start /Users/pdehaan/dev/tmp/hapi-test-5
> node server.js


/Users/pdehaan/dev/tmp/hapi-test-5/node_modules/hapi/node_modules/hoek/lib/index.js:425
    throw new Error(msgs.join(' ') || 'Unknown error');
          ^
Error: Invalid server options  {
  "maxSockets": null,
  "router": {
    "isCaseSensitive": true,
    "stripTrailingSlash": false
  },
  "state": {
    "cookies": {
      "parse": true,
      "failAction": "error",
      "clearInvalid": false,
      "strictHeader": true
    }
  },
  "location": "",
  "payload": {
    "maxBytes": 1048576,
    "uploads": "/var/folders/qc/g_f0szh95k1fzph34rhmf5jw0000gp/T/"
  },
  "validation": null,
  "json": {
    "replacer": null,
    "space": null,
    "suffix": null
  },
  "files": {
    "relativeTo": ".",
    "etagsCacheMaxSize": 10000
  },
  "timeout": {
    "client": 10000,
    "server": false
  },
  "load": {
    "maxHeapUsedBytes": 0,
    "maxRssBytes": 0,
    "maxEventLoopDelay": 0,
    "sampleInterval": 0
  },
  "debug": {
    "request": [
      "implementation"
    ]
  },
  "labels": [],
  "cors": true,
  "security": false,
  "cache" [1, 4]: {
    "engine": "redis",
    "partition": "cm"
  }
}

[1] cache must be a Function
[2] engine must be an object
[3] engine must be a Function
[4] cache must be an array
    at Object.exports.assert (/Users/pdehaan/dev/tmp/hapi-test-5/node_modules/hapi/node_modules/hoek/lib/index.js:425:11)
    at Object.exports.assert (/Users/pdehaan/dev/tmp/hapi-test-5/node_modules/hapi/lib/schema.js:15:10)
    at new module.exports.internals.Server (/Users/pdehaan/dev/tmp/hapi-test-5/node_modules/hapi/lib/server.js:75:12)
    at Object.exports.createServer (/Users/pdehaan/dev/tmp/hapi-test-5/node_modules/hapi/lib/index.js:14:12)
    at Object.<anonymous> (/Users/pdehaan/dev/tmp/hapi-test-5/server.js:16:19)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)

npm ERR! foo@0.0.1 start: `node server.js`
npm ERR! Exit status 8
npm ERR!
npm ERR! Failed at the foo@0.0.1 start script.
npm ERR! This is most likely a problem with the foo package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node server.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls foo
npm ERR! There is likely additional logging output above.
npm ERR! System Darwin 12.5.0
npm ERR! command "node" "/usr/local/bin/npm" "start"
npm ERR! cwd /Users/pdehaan/dev/tmp/hapi-test-5
npm ERR! node -v v0.10.30
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     /Users/pdehaan/dev/tmp/hapi-test-5/npm-debug.log
npm ERR! not ok code 0
```

---

**UPDATE:** The workaround seems to be changing from this (server.js#L16):

``` js
// var server = Hapi.createServer('0.0.0.0', config.port, options);
```

... to this: 

``` js
var server = Hapi.createServer('0.0.0.0', config.port, { cors: true });
```

**Comment by pdehaan** [comment]
My generated package.json file isn't very exciting, but looks like this:

``` json
{
  "name": "foo",
  "version": "0.0.1",
  "description": "bar",
  "author": "me",
  "licenses": "BSD",
  "repository": {
    "type": "git",
    "pathPrefix": ""
  },
  "dependencies": {
        "hapi": "latest",
        "catbox": "latest",
        "joi": "latest",
        "lodash": "latest",
        "q": "latest",
        "redis": "latest",
        "request": "latest",
        "winston": "latest"
},
  "devDependencies": {
    "grunt": "~0.4.1",
    "grunt-contrib-jshint": "~0.6.0",
    "grunt-contrib-watch": "~0.4.4",
    "grunt-jasmine-node": "~0.1.0",
    "grunt-jasmine-node-coverage": "~0.1.6",
    "grunt-jasmine-node-coverage-validation": "0.0.4",
    "grunt-ngdocs": "~0.1.8",
    "grunt-notify": "~0.2.17",
    "grunt-plato": "~0.2.1",
    "istanbul": "~0.1.44",
    "load-grunt-tasks": "~0.2.0",
    "rewire": "~1.1.3",
    "sinon": "~1.7.2",
    "supervisor": "~0.5.2"
  },
  "engines": {
    "node": ">=0.10.25"
  }
}
```

**Comment by pdehaan** [comment]
Installed modules + version numbers are:

``` sh
$ npm ls --depth 0
foo@0.0.1 /Users/pdehaan/dev/tmp/hapi-test-5
├── catbox@3.4.2
├── grunt@0.4.5
├── grunt-contrib-jshint@0.6.5
├── grunt-contrib-watch@0.4.4
├── grunt-jasmine-node@0.1.0
├── grunt-jasmine-node-coverage@0.1.11
├── grunt-jasmine-node-coverage-validation@0.0.4
├── grunt-ngdocs@0.1.11
├── grunt-notify@0.2.20
├── grunt-plato@0.2.1
├── hapi@6.8.1
├── istanbul@0.1.46
├── joi@4.7.0
├── load-grunt-tasks@0.2.1
├── lodash@2.4.1
├── q@1.0.1
├── redis@0.12.1
├── request@2.44.0
├── rewire@1.1.3
├── sinon@1.7.3
├── supervisor@0.5.7
└── winston@0.8.0
```

**Comment by maxwellito** [comment]
This looks like the default templates for hapi routes needs to be updated for the last version. 
Hapi is making new version of their framework quite regulary, the generator has to follow the latest API.

Question, why do you use 'npm start' when no rules for 'start' are defined in your 'package.json'?

**Comment by pdehaan** [comment]
Sorry, force of habit (`npm start`) as I always set it up in my projects (and `npm test`).
And by complete coincidence, it looks like `node server.js` happens to be the default behavior ([source](https://www.npmjs.org/doc/misc/npm-scripts.html#default-values)).

**Comment by maxwellito** [comment]
Oh nice, good to know. I thought the default behavior would be `node index.js`. Thanks for the discovery.
Once I'm back home (in 10 days :-S), I'll make a pull request to fix it. I need to talk with @lefos987 to update this generator.
Thanks for your issue (with the fix), that will help me.
I'll come back to you once updated.

**Comment by lefos987** [comment]
Closed by #18
